funsor/distribution.py

```diff
@@ -249,23 +249,30 @@ def _infer_param_domain(cls, name, raw_shape):
         # `infer_param_domain` methods.
         # Because NumPyro and Pyro have the same pattern, we use name check for simplicity.
         support_name = type(support).__name__
+
+        event_dim = 0
+        while support_name == "_IndependentConstraint":
+            event_dim += support.event_dim
```

Comment:
For now this should be corrected to
```py
event_dim += support.reinterpreted_batch_ndims
```
In the future (after https://github.com/pytorch/pytorch/pull/50547) we should eliminate this computation and simply define at the top
```py
event_dim = constraint.event_dim
```